typing/typedecl.ml

```diff
@@ -238,9 +293,35 @@ let transl_declaration env sdecl id =
       transl_simple_type env false sty', loc)
     sdecl.ptype_cstrs
   in
+  let add_attribute =
+    (* If the type is unboxable, check the attributes and add "unboxed"
+       if appropriate. *)
+    match sdecl.ptype_kind with
+    | Ptype_variant [{pcd_args = Pcstr_tuple [_]; _}]
+      | Ptype_variant [{pcd_args = Pcstr_record
+                          [{pld_mutable = Immutable; _}]; _}]
+      | Ptype_record [{pld_mutable = Immutable; _}] ->
+    let unboxed = Builtin_attributes.has_unboxed sdecl.ptype_attributes in
+    let boxed = Builtin_attributes.has_boxed sdecl.ptype_attributes in
+    if boxed && unboxed then raise (Error(sdecl.ptype_loc, Boxed_and_unboxed))
+    else if boxed then []
+    else if unboxed then []
+    else if !Clflags.unboxed_types then
+      [ ({txt = "ocaml.unboxed (default)"; loc = sdecl.ptype_loc}, PStr []) ]
+    else
+      [ ({txt = "ocaml.boxed (default)"; loc = sdecl.ptype_loc}, PStr []) ]
+        (* Note that we add the "boxed" annotation for symmetry,
+           but in fact we don't use it. *)
+    | _ -> []
+  in
+  let attributes = add_attribute @ sdecl.ptype_attributes in
```

Comment:
Do these synthesized attributes (with "(default)") end up in the .cmt / .cmi files?

Normally, the attribute identifier should follow a specific syntactic form (which excludes in particular whitespace and parenthesis characters), and tools (even Pprintast) could be disturbed by those names if they are not removed later.  And if they are removed later, is it relevant to distinguish the default/non-default cases.  Anyway, if this is purely for internal use and not for external tools, it does not seem very good to pass the information through attributes.